src/committee/orchestrator.rs

```diff
@@ -347,17 +335,25 @@ impl Orchestrator {
                     Ok(x) => x,
                     Err(rpc_error) => {
                         warn!("Round 1 error with {}, marking as disconnected and retrying round 1: {rpc_error}", member.address);
-                        let mut ms_w = member_status.write().unwrap();
+                        let mut ms_w = self.member_status.write().unwrap();
                         ms_w.mark_as_disconnected(member_id);
                         continue 'retry;
                     }
                 };
 
-                let response: bitcoincore_rpc::jsonrpc::Response = serde_json::from_str(&resp)?;
-                let resp: Round1Response = response.result()?;
+                if let Ok(response) =
+                    serde_json::from_str::<bitcoincore_rpc::jsonrpc::Response>(&resp)
+                {
+                    let resp: Round1Response = response.result()?;
```

Comment:
I'm wondering if it would be cleaner to do something like this:

```rust
let res = resp.map_err(|e| format!("connection error: {e}").and_then(|r| serde_json::from_str::< bitcoincore_rpc::jsonrpc::Response>(&r).map_err(|e| format!("json error: {e}"))).and_then(|r| r.result().map_err(|e| format!("rpc error: {e}")));
let resp = match res {
  Ok(x) => x,
  Err(e) => {
     warn!("Round 1 error with {addr}: {e}", addr=member.address);
     self.member_status.write().unwrap().mark_as_disconnected(member_id);
     need_retry = true // since at this point we have waiting for all the response might as well check all of them before retrying (so that we potentially discover multiple disconnects?)
  }
}
```

Reply:
Yep, saw that. We won't even need the `response.result()?` part once we have a custom serde struct for Node RPC responses. I can open up an issue for this, but would rather fix this after we have at least *some* e2e tests so I can just run the job and make sure I don't broken anything by accident, without having to recreate the whole environment each time.